### PR TITLE
VariableManager: Faster input_variables().

### PR DESCRIPTION
**Pull request by ltfish**
Cache variables without write accesses to improve the speed of input_variables().

This PR reduces the variable recovery time on `444a:sub_18013afb0` from 34 seconds to 10 seconds.

### CODE REVIEW

File: angr/knowledge_plugins/variables/variable_manager.py [hunk 1]
```diff
@@ -693,7 +702,7 @@ def get_phi_variables(self, block_addr):
             variables[phi] = self._phi_variables[phi]
         return variables
 
-    def input_variables(self, exclude_specials=True):
+    def get_variables_without_writes(self) -> List[SimVariable]:
```
Comment (xxr0ss) [inline angr/knowledge_plugins/variables/variable_manager.py:705]: Returned list here, but is used as a set elsewhere, Maybe the correct return value should be an OrderedSet?

File: angr/knowledge_plugins/variables/variable_manager.py [hunk 2]
```diff
@@ -379,6 +385,7 @@ def set_variable(self, sort, start, variable: SimVariable):
             pass
         region.set_variable(start, variable)
         self._variables.add(variable)
+        self._variables_without_writes.add(variable)
```
Comment (xxr0ss) [inline angr/knowledge_plugins/variables/variable_manager.py:388]: `_variables_without_writes` used as a set

**Review by xxr0ss — COMMENTED** [review]
```
File "/angr/angr/knowledge_plugins/variables/variable_manager.py", line 386, in set_variable
    self._variables_without_writes.add(variable)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'add'
```

File: angr/knowledge_plugins/variables/variable_manager.py (hunk 1 above)
Comment (ltfish) [reply angr/knowledge_plugins/variables/variable_manager.py:705]: Thanks for the bug report.

Comment (ltfish) [reply angr/knowledge_plugins/variables/variable_manager.py:705]: https://github.com/angr/angr/commit/d8c1b3a0b4c49de014319a96862705b4be348dc9 fixes this issue.

**Pull request merged**